src/Microsoft.AspNetCore.Html.Abstractions/HtmlContentBuilder.cs

```diff
@@ -97,6 +97,60 @@ public IHtmlContentBuilder Clear()
         }
 
         /// <inheritdoc />
+        public void CopyTo(IHtmlContentBuilder destination)
+        {
+            for (var i = 0; i < Entries.Count; i++)
+            {
+                var entry = Entries[i];
+
+                string entryAsString;
+                IHtmlContentContainer entryAsContainer;
+                if ((entryAsString = entry as string)  != null)
+                {
```

Comment:
nit: Is a bit ugly to do the assignment inside the if condition, we don't do it through the code base. I don't expect a side effect happening while checking a condition